src/core/server/plugins/plugins_system.test.ts

```diff
@@ -116,7 +116,7 @@ test('`setupPlugins` throws plugin has missing required dependency', async () =>
   pluginsSystem.addPlugin(createPlugin('some-id', { required: ['missing-dep'] }));
 
   await expect(pluginsSystem.setupPlugins(setupDeps)).rejects.toMatchInlineSnapshot(
-    `[Error: Topological ordering of plugins did not complete, these edges could not be ordered: [["some-id",{}]]]`
+    `[Error: Topological ordering of plugins did not complete, cyclic dependency detected between: ["some-id"]]`
```

Comment:
No, this message is for cyclic dependencies, not missing ones.

Reply:
but we test here 
```
`setupPlugins` throws plugin has missing required dependency` 
```

Reply:
I apologize, you are actually right. The error can also be thrown in case of missing dependencies. 

However, in real case usage, it will never occurs as `PluginService#shouldEnablePlugin` will return false and therefor not adding the plugin to the pluginSystem, so `PluginSystem#getTopologicallySortedPluginNames` will never be called with missing deps.

So not sure how what is the best thing to do here. Should I remove the unit test because it's testing a non-existing scenario, or should I put a more generic error message like `these plugins has cyclic or missing dependencies: ${edgesLeft}`

I think the second option is still better. @restrry wdyt ?